finalize general-helpers.metta refactor to PeTTa

yabuz87 opened the pull request:
## Description
This refactor improves function naming consistency and resolves several compilation and compatibility issues within PeTTa. Key changes include renaming conflicting functions, fixing parsing errors, and ensuring smoother cross-file execution.  
This PR finalizes the `general-helpers.metta` refactoring, except for the pending `add-reduct` function.

## Motivation and Context
These updates were necessary to improve compatibility with PeTTa’s built-in functions, prevent Prolog parsing conflicts, and maintain consistent function behavior after translation. The changes enhance code clarity, maintainability, and reduce runtime errors.

## How Has This Been Tested?
Tested locally by reloading all affected modules and verifying successful compilation and execution of dependent functions. Confirmed that previous calling patterns remain compatible and all related tests pass.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (code improvement without changing external behavior)

## Checklist
- [x] My code follows the code style of this project.  
- [x] I have verified that renamed and refactored functions compile and execute correctly.  
- [x] All new and existing tests passed.

Yagth commented on utilities/general-helpers.metta:
```diff
@@ -683,7 +681,7 @@
 ;; applying the two functions in sequence.
 ;; (: . (-> (-> $b $c) (-> $a $b) (-> $a $c)))
-;; (= ((. $g $f) $x) ($g ($f $x)))
+;; (= ((. $g $f) $x) ($g ($f $x)));in petta . is internally used by Prolog for building lists and dicts and what it does is the same for compose function.
```
Do we have a built in function called `.`?

yabuz87 commented:
> Did you check if the functions renamed in this PR are also renamed everywhere they are used?

Yes, I verified that. I renamed the two functions —`` flatten`` to ``flatten-list`` and ``key`` to ``key-of``.
Both were not referenced in any other files after checking their occurrences, so no additional updates were required.
